mountpoint-s3/src/inode.rs

```diff
@@ -267,8 +267,13 @@ impl Superblock {
         logging::record_name(inode.name());
         let mut sync = inode.get_mut_inode_state()?;
 
+        // If the inode is remote, we would typically throw an error here, but instead we
+        // silently fail and return the current stat. This allows programs like `wget` to
+        // continue working.
         if sync.write_status == WriteStatus::Remote {
-            return Err(InodeError::SetAttrNotPermittedOnRemoteInode(inode.err()));
+            let stat = sync.stat.clone();
+            drop(sync);
```

Comment:
Is this `drop` needed to address a compiler complaint? It gets dropped when this `setattr` returns doesn't it?

Reply:
Yes it's for the compiler. Not sure how else to handle it.